src/graphs/ValidateBinaryTreeNodes.java

```diff
@@ -0,0 +1,135 @@
+package graphs;
+
+import java.util.Scanner;
+
+public class ValidateBinaryTreeNodes {
+
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        int node_number = sc.nextInt();
+
+        int[] left_child_array = new int[node_number];
+        int[] right_child_array = new int[node_number];
+
+        for (int i = 0; i < node_number; i++) {
+            left_child_array[i] = sc.nextInt();
+            right_child_array[i] = sc.nextInt();
+        }
+        System.out.println(validateBinaryTreeNodes(node_number, left_child_array, right_child_array));
+
+        sc.close();
+    }
+
+    private static boolean validateBinaryTreeNodes(int n, int[] leftChild, int[] rightChild) {
+        byte[] edges_count_array = new byte[n];
+        int[] parent_array = new int[n];
+        boolean root_found = false;
+
+        // start initialisation of parent array
+        for (int i = 0; i < n; i++) {
+            parent_array[i] = -1;
+        }
+
+        // check one node case
+        if (n == 1) {
+            return true;
+        }
+        for (int i = 0; i < n; i++) {
+            int left_child = leftChild[i];
+            int right_child = rightChild[i];
+
+            if (left_child > -1) {
+                parent_array[left_child] = i;
+                // check number of input edges for left child
+                if (edges_count_array[left_child] > 0) {
+                    return false;
+                }
+                edges_count_array[left_child]++;
+            }
+            if (right_child > -1) {
+                parent_array[right_child] = i;
+                // check number of input edges for right child
+                if (edges_count_array[right_child] > 0) {
+                    return false;
+                }
+                edges_count_array[right_child]++;
+            }
+        }
+
+        // check root consistence
+        for (int i = 0; i < n; i++) {
+            if (edges_count_array[i] == 0) {
+                // the root is already found so we have more then one tree
+                if (root_found) {
+                    return false;
+                }
+                root_found = true;
+            }
+        }
+        // if root don't found we have loop in the graph
+        return root_found && !check_loop(parent_array);
+    }
+
+    private static boolean check_loop(int[] parent_array) {
+        int size = parent_array.length;
+        boolean[] is_marked = new boolean[size];
+        int mark_count = 0;
+
+        int cur_idx = 0;
+        int start_idx = -1;
+        // state variable: 0 - is moving on parent links; 1 - move from marked node
+        int state = 0;
```

Comment:
```suggestion
        // state variable: 0 - is moving on parent links; 1 - move from marked node
        int state = 0;
```
refactor comment for correct description of states